server/market/market.go

```diff
@@ -587,6 +588,79 @@ func (m *Market) SubmitOrder(rec *orderRecord) error {
 	return <-m.SubmitOrderAsync(rec)
 }
 
+// processCancelOrderWhileSuspended is called when cancelling an order while
+// the market is suspended and Run is not runnning.
+//
+// This function:
+// 1. Removes the target order from the book.
+// 2. Unlocks the order coins.
+// 3. Updates the storage with the new cancel order and cancels the existing limit order.
+// 3. Responds to the client that the order was recieved.
+// 4. Sends the unbooked order to the order feeds.
+// 5. Creates a match object and sends it to the swapper to notify the client of the match.
+func (m *Market) processCancelOrderWhileSuspended(rec *orderRecord) error {
+	co, ok := rec.order.(*order.CancelOrder)
+	if !ok {
+		return ErrInvalidOrder
+	}
+
+	lo, ok := m.book.Remove(co.TargetOrderID)
```

Comment:
Yeah our general convention for mutexes is to have the fields of a struct grouped by what mutex is intended to guard the fields.  In this case @JoeGruffins is right that the `settling` map does need the mutex locked.  I'm less sure about calling `book.Remove` at the moment because it has it's own internal synchronization mechanisms, but holding `bookMtx` for both ops is fine.

I'm pretty sure in most places when we lock `bookMtx` around accesses to the `book` field, we are really trying preform other operations with a consistent book view, rather than avoid a data race, but I do see some discrepancies.  I'll look closer.